### DISCUSSION TITLE
Self-hosted Gateway Failure

### DISCUSSION THREAD

**Discussion by chaseconey** [codecov/feedback#155] [Q&A] [answered]
It seems that after the latest changes that went out on the gateway, the self-hosted setup is no longer working.

Getting this error with the latest of all products:

```
self-hosted-gateway-1   | Still waiting for Codecov Default to start ...
self-hosted-gateway-1   | Timeout waiting for Codecov Default to start
self-hosted-gateway-1   | Review documentation: https://docs.codecov.io/docs/configuration
```

I have looked around at the new environment variables and tried various combinations of them and nothing seems to get the gateway to pick up the default endpoint like it should.

Any ideas what could be missing?

**Comment by tnt-sentry** [comment, marked as answer]
This is fixed by https://github.com/codecov/codecov-gateway/pull/80

The latest rolling release will have it and the next monthly release. In the interim of the release, you can also fix by specifying the env vars from that PR on your gateway deployment.

```
CODECOV_DEFAULT_HOST=frontend
CODECOV_DEFAULT_PORT=8080
CODECOV_DEFAULT_SCHEME=http
CODECOV_DEFAULT_HOST_HEADER="%[req.hdr(Host)]"
```

**Reply by chaseconey** [reply]
This did it! Thanks Trent!
